### PR TITLE
separate cmake-compilation into different travis task

### PR DESCRIPTION
**Pull request by zyearn**

### CODE REVIEW

File: build_in_travis_ci.sh [hunk 1]
```diff
@@ -21,39 +21,30 @@ runcmd(){
...
-echo "start building by cmake"
-rm -rf bld && mkdir bld && cd bld
 if [ "$PURPOSE" = "compile" ]; then
-    if ! cmake ..; then
-        echo "Fail to generate Makefile by cmake"
-        exit 1
-    fi
-    make -j4
+    init_make_config
+    make -j4 #&& sh tools/make_all_examples
```
Comment (zyearn) [inline build_in_travis_ci.sh]: Thrift example挂了，这里把make_all_examples注释掉。之前没挂的原因是紧接着cmake编译成功了，所以脚本的退出码是0。另外单独开个pr解决Thrift example编译问题吧

**Comment by jasonszang** [comment]
Hi! Is there something wrong with mesalink currently? My PR fails because of unabling to build mesalink and I for sure didn't touch anything near it.
Is this related to this PR?

File: build_in_travis_ci.sh (hunk 1 above)
Comment (jamesge) [reply build_in_travis_ci.sh]: 就在这个里面一并解决吧，不要注来注去了

File: .travis.yml [hunk 2]
```diff
@@ -25,8 +26,7 @@ install:
 - sudo apt-get install -qq realpath libgflags-dev libprotobuf-dev libprotoc-dev protobuf-compiler libleveldb-dev libgoogle-perftools-dev libboost-dev libssl-dev libevent-dev libboost-test-dev
 - sudo apt-get install libgtest-dev && cd /usr/src/gtest && sudo env "PATH=$PATH" cmake . && sudo make && sudo mv libgtest* /usr/lib/ && cd -
 - sudo apt-get install -y gdb  # install gdb
-- if [[  "$USE_MESALINK" == "yes" ]]; then curl https://sh.rustup.rs -sSf | sh -s -- -y && source $HOME/.cargo/env && wget https://github.com/mesalock-linux/mesalink/archive/v0.8.0.tar.gz && tar -xf v0.8.0.tar.gz && cd mesalink-0.8.0 && ./autogen.sh --prefix=/usr/  && make && sudo make install && cd - ; fi
+#- if [[  "$USE_MESALINK" == "yes" ]]; then curl https://sh.rustup.rs -sSf | sh -s -- -y && source $HOME/.cargo/env && wget https://github.com/mesalock-linux/mesalink/archive/v0.8.0.tar.gz && tar -xf v0.8.0.tar.gz && cd mesalink-0.8.0 && ./autogen.sh --prefix=/usr/  && make && sudo make install && cd - ; fi
```
Comment (jamesge) [inline .travis.yml]: 这行不用动吧

Comment (zyearn) [reply .travis.yml]: 就是这行挂了

**Comment by zyearn** [comment]
> Hi! I saw you did a test pr and then commented out mesalink from travis config. Is there something wrong with mesalink currently? My PR fails because of unabling to build mesalink and I for sure didn't touch anything near it.

Yes. Brpc can't be built in travis since some build problem in mesalink. I have reminded the author to solve the issue.

File: build_in_travis_ci.sh (hunk 1 above)
Comment (zyearn) [reply build_in_travis_ci.sh]: done

**Comment by jasonszang** [comment]
Can we depend on a pre-built mesalink on travis instead? Is it necessary that we build mesalink from source?

File: build_in_travis_ci.sh [hunk 3]
```diff
@@ -21,39 +21,30 @@ runcmd(){
...
 
+init_make_config() {
 EXTRA_BUILD_OPTS=""
 if [ "$USE_MESALINK" = "yes" ]; then
     EXTRA_BUILD_OPTS="$EXTRA_BUILD_OPTS --with-mesalink"
 fi
 
 # The default env in travis-ci is Ubuntu.
-if ! sh config_brpc.sh --headers=/usr/include --libs=/usr/lib --nodebugsymbols --cxx=$CXX --cc=$CC $EXTRA_BUILD_OPTS; then
+if ! sh config_brpc.sh --headers=/usr/include --libs=/usr/lib --nodebugsymbols --cxx=$CXX --cc=$CC $EXTRA_BUILD_OPTS --with-thrift ; then
```
Comment (jamesge) [inline build_in_travis_ci.sh]: 为什么要with-thrift?

Comment (zyearn) [reply build_in_travis_ci.sh]: 因为thrift example需要依赖开这个选项的brpc lib。example只有在"$PURPOSE" = "compile"的时候跑，或许这个with-thrift可以放到EXTRA_BUILD_OPTS里，我改一下

**Pull request merged**